scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java

```diff
@@ -285,162 +318,241 @@ protected void logDataspacesStatus(String message, DataspacesStatusLevel level)
-                org.objectweb.proactive.extensions.dataspaces.vfs.selector.FileSelector selector =
-                        new org.objectweb.proactive.extensions.dataspaces.vfs.selector.FileSelector();
-                selector.setIncludes(is.getInputFiles().getIncludes());
-                selector.setExcludes(is.getInputFiles().getExcludes());
+            if (exception != null) {
+                throw exception;
+            }
 
-                logger.debug("Selector used is " + selector);
+            String inputSpaceUri = virtualResolve(INPUT);
+            String outputSpaceUri = virtualResolve(OUTPUT);
+            String globalSpaceUri = virtualResolve(GLOBAL);
+            String userSpaceUri = virtualResolve(USER);
 
-                switch (is.getMode()) {
-                    case TransferFromInputSpace:
-                        toBeThrown =
-                                copyInputDataToScratch(INPUT, "INPUT", is, selector, inResults);
-                        break;
-                    case TransferFromOutputSpace:
-                        toBeThrown =
-                                copyInputDataToScratch(OUTPUT, "OUTPUT", is, selector, outResults);
-                        break;
-                    case TransferFromGlobalSpace:
-                        toBeThrown =
-                                copyInputDataToScratch(GLOBAL, "GLOBAL", is, selector, globResults);
-                        break;
-                    case TransferFromUserSpace:
-                        toBeThrown =
-                                copyInputDataToScratch(USER, "USER", is, selector, userResults);
-                    case none:
-                        //do nothing
-                        break;
-                }
-            }
+            Map<String, DataSpacesFileObject> filesToCopy =
+                    createFolderHierarchySequentially(
+                            inputSpaceUri, inputSpaceFiles,
+                            outputSpaceUri, outputSpaceFiles,
+                            globalSpaceUri, globalSpaceFiles,
+                            userSpaceUri, userSpaceFiles);
 
-            if (toBeThrown != null) {
-                throw toBeThrown;
-            }
+            List<Future<Boolean>> transferFutures =
+                    doCopyInputDataToScratchSpace(filesToCopy);
 
-            String outuri = (OUTPUT == null) ? "" : OUTPUT.getVirtualURI();
-            String globuri = (GLOBAL == null) ? "" : GLOBAL.getVirtualURI();
-            String useruri = (USER == null) ? "" : USER.getVirtualURI();
-            String inuri = (INPUT == null) ? "" : INPUT.getVirtualURI();
+            handleResults(transferFutures);
+        } finally {
+            // display dataspaces error and warns if any
+            displayDataspacesStatus();
+        }
+    }
 
-            Set<String> relPathes = new HashSet<>();
+    private Map<String, DataSpacesFileObject> createFolderHierarchySequentially(String inputSpaceUri,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 8 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS00107)